core-api/src/main/java/com/optimizely/ab/event/BatchEventProcessor.java

```diff
@@ -240,42 +230,64 @@ public static Builder builder() {
     public static class Builder {
         private BlockingQueue<Object> eventQueue = new ArrayBlockingQueue<>(DEFAULT_QUEUE_CAPACITY);
         private EventHandler eventHandler = null;
-        private Integer batchSize = null;
-        private Long flushInterval = null;
+        private Integer batchSize = PropertyUtils.getInteger(CONFIG_BATCH_SIZE, DEFAULT_BATCH_SIZE);
+        private Long flushInterval = PropertyUtils.getLong(CONFIG_BATCH_INTERVAL, DEFAULT_BATCH_INTERVAL);
+        private Long timeoutMillis = PropertyUtils.getLong(CONFIG_CLOSE_TIMEOUT, DEFAULT_TIMEOUT_INTERVAL);
         private ExecutorService executor = null;
         private NotificationCenter notificationCenter = null;
-        private Long timeoutMillis = null;
 
+        /**
+         * {@link EventHandler} implementation used to dispatch events to Optimizely.
+         */
         public Builder withEventHandler(EventHandler eventHandler) {
             this.eventHandler = eventHandler;
             return this;
         }
 
+        /**
+         * EventQueue is the underlying BlockingQueue used to buffer events before being added to the batch payload.
+         */
         public Builder withEventQueue(BlockingQueue<Object> eventQueue) {
             this.eventQueue = eventQueue;
             return this;
         }
 
+        /**
+         * BatchSize is the maximum number of events contained within a single event batch.
+         */
         public Builder withBatchSize(Integer batchSize) {
             this.batchSize = batchSize;
             return this;
         }
 
+        /**
+         * FlushInterval is the maximum duration, in milliseconds, that an event will remain in flight before
```

Comment:
nit. May be say event will remain in queue?

Reply:
I didn't want to confuse the pending batch with the `eventQueue`. There are no guarantees about how long an event might exist in the `eventQueue`.